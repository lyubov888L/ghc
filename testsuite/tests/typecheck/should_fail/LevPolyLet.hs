{-# language DataKinds #-}
{-# language KindSignatures #-}
{-# language PolyKinds #-}
{-# language RankNTypes #-}

module LevPolyLet
  ( example
  ) where

import GHC.Exts

-- This should be rejected because of the let binding.
example :: forall (v :: Levity) (a :: TYPE ('BoxedRep v)).
     (Int -> a)
  -> (a -> Bool)
  -> Bool
example f g =
  let x = f 42
   in g x
