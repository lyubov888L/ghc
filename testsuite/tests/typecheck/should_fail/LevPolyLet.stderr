LevPolyLet.hs:18:7:
    A levity-polymorphic type is not allowed here:
      Type: a
      Kind: TYPE ('BoxedRep v)
    In the type of binder ‘x’
